front/pages/api/stripe/webhook.ts

```diff
@@ -446,14 +447,51 @@ async function handler(
                 },
               });
             }
-            await subscription.update({ endDate });
+            await subscription.update({
+              endDate,
+              // if the subscription is canceled, we set the requestCancelAt date to now.
+              // if the subscription is reactivated, we unset the requestCancelAt date.
+              requestCancelAt: endDate ? now : null,
+            });
             const auth = await Authenticator.internalBuilderForWorkspace(
               subscription.workspace.sId
             );
             if (!endDate) {
               // Subscription is re-activated, so we need to unpause the connectors in case they were paused.
               await unpauseAllConnectorsAndCancelScrub(auth);
+
+              ServerSideTracking.trackSubscriptionReactivated({
+                workspace: renderLightWorkspaceType({
+                  workspace: subscription.workspace,
+                }),
+              }).catch((e) => {
+                logger.error(
+                  {
+                    panic: true,
```

Comment:
bearish on having a panic here. Not sure what I would do as eng runner here?

Reply:
It's just that I wanted to make sure I was aware of it, but agree that it's not actionable for the runner